docs/latest/design/extending-envoy-gateway.md

```diff
@@ -0,0 +1,302 @@
+- Extension developers should **NOT** have to maintain a custom fork of Envoy Gateway
+- Provide a system for extending Envoy Gateway which allows extension projects to ship updates independent of Envoy Gateway's release schedule
+- An Extension can hook into both the Envoy configuration pipeline and the infra manager pipeline
+- Modify the generated Envoy xDS and Envoy Gateway infra config (either directly via an xDS hook or earlier in the pipeline or by dynamically adding resources)
+- Setup a foundation for the initial iteration of Extending Envoy Gateway
+
+## Non-Goals
+
+- The initial design does not capture every hook that Envoy Gateway will eventually support.
+- Extend [Gateway API Policy Attachments][]. At some point, these will be addressed using this extension system, but the initial implementation omits these.
+- Support multiple extensions at the same time. Due to the fact that extensions will be modifying xDS resources after they are generated, handling the order of extension execution for each individual hook point is a challenge. Additionally, there is no 
+real way to prevent one extension from overwriting or breaking modifications to xDS resources that were made by another extension that was executed first.
+
+## Overview
+
+Envoy Gateway can be extended by vendors by means of an extension server developed by the vendor and deployed alongside Envoy Gateway.
+An extension server can make use of one or more pre/post hooks inside Envoy Gateway before and after its major components (translator, etc.) to allow the extension to modify the data going into or coming out of these components.
+An extension can be created external to Envoy Gateway as its own Kubernetes deployment or loaded as a sidecar. gRPC is used for the calls between Envoy Gateway and an extension. In the hook call, Envoy Gateway sends data as well 
+as context information to the extension and expects a reply with a modified version of the data that was sent to the extension. Since extensions fundamentally alter the logic and data that Envoy Gateway provides, Extension projects assume responsibility for any bugs and issues 
+they create as a direct result of their modification of Envoy Gateway.
+
+## Diagram
+
+![Architecture](../images/extension-example.png)
+
+## Registering Extensions in Envoy Gateway
+
+Information about the extension that Envoy Gateway needs to load is configured in the Envoy Gateway config. The extension must specify:
+
+- Which API groups the extension supports
+- The `Kind`s supported under each API group
+- Extension service connection info (host, port, TLS settings, etc.)
+
+An example configuration:
+
+```yaml
+apiVersion: config.gateway.envoyproxy.io/v1alpha1
+kind: EnvoyGateway
+extension:
+  apiGroups:
+  - apiGroup: example.myextension.io
+    apiVersion: v2
+    kinds:
+    - OAuth2Filter
+  service:
+    host: my-extension.example
+    port: 443
+    tls:
```

Comment:
can this look similar to https://gateway-api.sigs.k8s.io/references/spec/#gateway.networking.k8s.io%2fv1beta1.GatewayTLSConfig ?